shared/src/main/kotlin/org/jetbrains/kotlin/konan/target/Apple.kt

```diff
@@ -63,19 +64,35 @@ class AppleConfigurablesImpl(
         XcodePartsProvider.InternalServer
     } else {
         val xcode = Xcode.current
-        properties.getProperty("useFixedXcodeVersion")?.let { requiredXcodeVersion ->
-            val currentXcodeVersion = xcode.version
 
-            if (properties.getProperty("ignoreXcodeVersionCheck") != "true" &&
-                    currentXcodeVersion != requiredXcodeVersion) {
-                error("expected Xcode version $requiredXcodeVersion, got $currentXcodeVersion, consider updating " +
-                        "Xcode or use \"ignoreXcodeVersionCheck\" variable in konan.properties")
+        if (properties.getProperty("ignoreXcodeVersionCheck") != "true") {
+            properties.getProperty("minimalXcodeVersion")?.let { minimalXcodeVersion ->
+                val currentXcodeVersion = xcode.version
+                checkXcodeVersion(minimalXcodeVersion, currentXcodeVersion)
             }
         }
 
         XcodePartsProvider.Local(xcode)
     }
 
+    private fun checkXcodeVersion(minimalVersion: String, currentVersion: String) {
+        val minimalVersionParts = minimalVersion.split('.').map { it.toInt() }
+        val currentVersionParts = currentVersion.split('.').map { it.toInt() }
+        val size = max(minimalVersionParts.size, currentVersionParts.size)
+
+        for (i in 0 until size) {
+            val currentPart = currentVersionParts.getOrElse(i) { 0 }
+            val minimalPart = minimalVersionParts.getOrElse(i) { 0 }
+
+            when {
+                currentPart > minimalPart -> return
+                currentPart < minimalPart ->
+                    error("Unsupported Xcode version. Consider updating Xcode. " +
+                        "Minimal supported Xcode version is $minimalVersion, got $currentVersion.")
```

Comment:
Minimal supported Xcode version is $minimalVersion, got $currentVersion.->  Unsupported Xcode version $currentVersion, minimal supported version is $minimalVersion.

Reply:
Fixed.